File: src/main/java/com/fluxchess/pulse/CastlingType.java
/*
 * Copyright (C) 2013-2015 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */
package com.fluxchess.pulse;

final class CastlingType {

  static final int KINGSIDE = 0;
  static final int QUEENSIDE = 1;

  static final int NOCASTLINGTYPE = 2;

  static final int[] values = {
      KINGSIDE, QUEENSIDE
  };

  private CastlingType() {
  }

}

File: pulse-kotlin/src/commonMain/kotlin/com/fluxchess/pulse/kotlin/CastlingType.kt
/*
 * Copyright 2013-2024 Phokham Nonava
 *
 * Use of this source code is governed by the MIT license that can be
 * found in the LICENSE file.
 */

package com.fluxchess.pulse.kotlin

typealias CastlingType = Int

const val KINGSIDE: CastlingType = 0
const val QUEENSIDE: CastlingType = 1

const val NO_CASTLING_TYPE: CastlingType = 2

val castlingTypes = intArrayOf(KINGSIDE, QUEENSIDE)
